torch/distributed/_functional_ops.py

```diff
@@ -0,0 +1,31 @@
+import torch
+
+c10_lib = torch.library.Library("c10d_functional", "DEF")
+c10_lib_impl = torch.library.Library("c10d_functional", "IMPL")
+
+ops_defs = [
+    "all_reduce(Tensor self, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor",
+    "wait_tensor(Tensor self) -> Tensor",
+    "all_gather_into_tensor(Tensor shard, str tag, int[] ranks, int group_size) -> Tensor",
+    "reduce_scatter_tensor(Tensor input, str reduceOp, str tag, int[] ranks, int group_size) -> Tensor",
+]
+
+def _register_ops(resolve_ops):
+    import torch.distributed._functional_collectives as fun_col
+    my_module = fun_col
```

Comment:
shouldn't fun_col only be imported in the `if torch.distributed.is_available() True` case?  It looks like you'll import it even if not available here.  I think you meant to delete those 2 lines above?

Reply:
Import inside functions are only evaluated when the function is first called. That's why this works. :)

I removed this as I simplified the change even further.